### PR TITLE
create change status button and add turbolinks-compatibility

### PR DESCRIPTION
**Pull request by nooralnabi**
add turbolinks-compatibility js  to work all js with turbolinks fine 
add flash_types to application_controller to  show  messages when status is change

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,6 +34,8 @@ end
 
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 gem 'haml', '~> 5.0', '>= 5.0.1'
+gem 'jquery-turbolinks'
```
Comment (ssharba) [inline Gemfile]: try to make it work without 'jquery-turbolinks'

File: app/assets/javascripts/turbolinks-compatibility.coffee [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{defer, dispatch} = Turbolinks
```
Comment (ssharba) [inline app/assets/javascripts/turbolinks-compatibility.coffee]: @nooralnabi do we really need this?

File: app/controllers/contractors_controller.rb [hunk 3]
```diff
@@ -12,6 +12,17 @@ def index
   def show
   end
 
+
+  def change_status
+  @contractor = Contractor.find( params[:id])
+  @contractor.active =!@contractor.active
+  respond_to do |format|
+             if @contractor.save
```
Comment (ssharba) [inline app/controllers/contractors_controller.rb]: I would take this condition out of the response to be able to display error.

File: app/views/contractors/index.html.haml [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
         %th
     %tbody
       - @contractors.each do |contractor|
         %tr
           %td= contractor.contractor_id
           %td= contractor.client_id
-          %td= contractor.active
+          %td
+            - if contractor.active
+              %input{:checked => "checked",:class =>"contractor_check",:id=>"#{contractor.id}", "data-toggle" => "toggle",:onclick=> "change_status(#contractor.id)", "data-on" => "Active" , "data-off" => "Deactive" , :type => "checkbox"}/
```
Comment (ssharba) [inline app/views/contractors/index.html.haml]: I would show the end-user "active" and "inactive"

File: Gemfile (hunk 1 above)
Comment (nooralnabi) [reply Gemfile]: The toggle will not working with it  and compatibility.coffee is Necessary to

File: app/assets/javascripts/turbolinks-compatibility.coffee (hunk 2 above)
Comment (nooralnabi) [reply app/assets/javascripts/turbolinks-compatibility.coffee]: Its load faster  from  add turbolinks:load before your code in js

**Pull request merged**